### ISSUE TITLE
Suggesting alternate sites after March 31

### ISSUE THREAD

**Issue by ccd0** [ccd0/4chan-x#1539]
> Dear Webmaster,
> 
> You are receiving this email because you are registered as a website administrator using reCAPTCHA, and your website is still using reCAPTCHA v1, which will be turned off on March 31, 2018.
> 
> We announced the reCAPTCHA v1 deprecation in May 2016. Starting in November 2017, a small percentage of reCAPTCHA v1 traffic will begin to show a notice informing users that the old API will soon be retired. Any calls to the v1 API will not work after March 31, 2018.
> 
> To ensure continued functionality, you’ll need to update your website to a current version of reCAPTCHA. You can learn more about reCAPTCHA v2, Invisible reCAPTCHA and reCAPTCHA Android API in our Developer’s Guide. The new APIs are simple to implement and will streamline the captcha experience for your users. If you need help, you can engage in the reCAPTCHA Google Developer Group or post to Stack Overflow with the ‘recaptcha’ tag.
> 
> We hope that your upgrade will be seamless, and we’re confident you’ll be happy with the results.
> 
> Thank you,
> 
> reCAPTCHA Support

On March 31, 4chan will suffer a serious decline in usability. When that happens, some fraction of 4chan's users will get fed up and leave. It probably won't be significant from 4chan's perspective, but could be massive for alternative sites which are very lucky to have a tenth of 4chan's users. What's more, this exodus will be different from many previous exoduses because the users leaving will be higher quality posters than the average anon, rather than low-quality posters leaving because they got banned. After all, the new and the clueless are unlikely to even know that the option to use legacy captcha exists.

I think 4chan X should help the process along by recommending alternate sites. Recommendations could be shown in place of the text captcha in cases where it fails to load. Different recommendations could be shown for different boards.

To do this, I need to compile a list of boards that are willing, ready, and deserving to take a large influx of slightly above-average quality 4chan refugees. Preferably not 8ch-style board hosting services or sites with a reputation for raids on 4chan. I am neither seeking nor accepting compensation; these are not paid ads.

**Comment by mewch** [comment]
Hello ccd0,

I am the admin of rising, stable image board. I would be interested in being a part of your recommended alt chans. With strong confidence I'd be able to handle an influx in traffic. My image board is run by caring and experienced administrators. It would be a pleasure to be considered an alternative image board. Thank you.

- Mega Milk

**Comment by lc-guy** [comment]
^ mewch was regularly spammed on /soc/ and got urlbanned as a result, by the way. If you decide to add it, limit it to the soc-ey boards like /r9k/ and /soc/.

**Comment by ccd0** [comment]
> If you decide to add it, limit it to the soc-ey boards like /r9k/ and /soc/.

It would probably be best to limit all the sites to one or a few relevant boards.

**Comment by samachannel** [comment]
Hello! I received an email regarding this from a user.
I run Samachan, an imageboard based around Anime and Japanese culture.
We have two boards, Anime/Otaku and 2D/Everything (/a/ and /z/ are the directories)
We'd love to be put down as an alternative. Thanks!

**Comment by deysu** [comment]
Hi there, another mewch administrator. Don't mean to derail this thread into discussion of mewch, but some clarification could be used:

Mewch was never spammed on /soc/. You are confusing mewch for ChanPink, which _was_ advertised on /soc/. ChanPink was closed down quite some time ago by the other administrator Mega Milk. Although mewch was created by the same administrator, it was not spammed on 4chan and is currently not urlbanned. Mewch's top boards are /fringe/, a board that is /x/-like, and /b/. These boards are nothing like /soc/ or the earlier ChanPink's /campus/.

Agreed also on limiting of all sites to a few relevant boards. Quality users should be spread out far and wide, not limited to any one particular imageboard.

**Comment by antonizoon** [comment]
I don't understand how this change dissuades 4chan users from coming. In fact I don't see how it changes anything other than a few swaps in API calls. 

Even though 4chan allegedly uses ReCaptcha v1 APIs, **it has used Recaptcha v2 in the user facing side for years.**

This is how Google defines Recaptcha v1 and v2: https://support.google.com/recaptcha/?hl=en#6080933

ReCaptcha v1 (with 2 words) has not been in use on 4chan in years. Even when it was first introduced by Moot, it didn't really stem the user growth. For those who don't remember, this is what it looked like:

![](https://lh4.ggpht.com/Kzeyj9SroRcche4lG7gBAXpO38BHjSFDiT7KEXtwvOwWc5p7fSKHR0Vv8uWT=w895)

This is ReCaptcha v2. Looks familiar right? Because that's exactly what it looks like now, for more than a year. The userbase never dropped significantly here either.

![](https://lh5.ggpht.com/fCcxMjTcSQHBzGNIo-vmzScywQE-6aYakCMwy3o6QsoK6DxZsAETQpTpPEYZ=w616)

I would say that most of the original 4chan users have already moved on to other places (including, holy hell, Reddit, Tumblr, and Twitter), but 4chan keeps growing because it is 4chan: because it brings so many people and posts together in one place. 

Unless 4chan itself disappears, it cannot be replaced wholesale like Reddit did to Digg.

**Comment by lc-guy** [comment]
4chan X allows you to use legacy recaptcha.

**Comment by antonizoon** [comment]
How many 4chan users use 4chan X: and out of those, specifically for the purpose of using legacy recaptcha? 

I wouldn't say none, but it is not going to be the downfall of the entire community of 4chan.

**Comment by ccd0** [comment]
> I wouldn't say none, but it is not going to be the downfall of the entire community of 4chan.

I don't disagree with you.

> It probably won't be significant from 4chan's perspective, but could be massive for alternative sites which are very lucky to have a tenth of 4chan's users.

**Comment by lc-guy** [comment]
>it is not going to be the downfall of the entire community of 4chan.
I don't think the OP said that either.

>some fraction of 4chan's users will get fed up and leave. It probably won't be significant from 4chan's perspective

In addition, many people on the slower boards related to otaku culture use this feature, from what I've seen, although my personal experience doesn't hold much weight.

**Comment by ccd0** [comment]
The native extension also has a setting to use v1.

**Comment by antonizoon** [comment]
Perhaps the 4chan X script could be used to function as an Overchan in the spirit of this Github issue, which could recommend other imageboards that are in similar design and topic as the 4chan board.

**Comment by antonizoon** [comment]
Another related question is: if these users left simply because they liked ReCaptcha v1, how will captchas be handled on the smaller communities? ReCaptcha v1 can't be used there when it is discontinued.

What do we consider an acceptable captcha instead of ReCaptcha v1? What if the alternative imageboard also uses ReCaptcha v2, is that fine or is that not allowed? Without them, although there are fewer users, there will still remain a similar vulnerability to spam.

**Comment by ccd0** [comment]
ReCaptcha v1 does little to deter spam. It has been left in a weak state by Google for over a year, presumably to encourage people to upgrade.

Not all sites need to have captchas, or demand a captcha with every post. Even 4chan seems to have made the judgment that they only need real captcha protection for thread creation. It's good to look at other factors, such as how fast a user is making posts, or whether the post contains suspicious text such as hyperlinks, the HTTP headers, and the country or IP range being posted from, to decide whether to demand a captcha. Even using Recaptcha could be better than what 4chan does if it wasn't required for every post.

Still, it's a good question what sort of captcha people should use if they need to. I'm not sure if I have the answer. If I were in that situation, I would be inclined to roll my own. Because an attacker would have to break my captcha specifically, it would make the site less attractive to spam. Granted, it would probably be a weak captcha. And I've seen many people design captchas with serious obvious flaws, such as the old obfuscated CSS captcha 8chan used at one point, and the trivially OCR-able captcha on Endchan where they for some inscrutable reason draw colored blobs over the letters that interfere with humans reading it but not computers. But even the obfuscated CSS one required some effort for people to break it (the one on Endchan, not so much).

But more important than captcha is a system that prevents abuse even in the event a captcha is broken. In particular, it's good to limit the total thread creation rate (or at least the pruning rate) so spammers can't wipe the board, something 4chan has failed to do (although the archive alleviates the problem a bit).

**Comment by ccd0** [comment]
I don't think alternatives need to compete against the whole of 4chan. It's healthier to have lots of sites, each with their own niche.

**Comment by antonizoon** [comment]
So captchas aside, what do you plan to have the 4chan-X extension do for allowing users to try other imageboards (perhaps based on the topic of the board they are currently on)? If so, how will such alternative imageboards be rated?

**Comment by Fasmea** [comment]
I strongly disapprove of this. Although I personally use more sites than 4chan, I don't want to see any alternative or competing imageboards being advertised, as I feel it's against the spirit of free software.

**Comment by antonizoon** [comment]
Yes, I'm not really sold on the question of whether the abandonment of recaptcha v1 (which no one will be able to use thereafter anyway) will really be a major push for users to move to other imageboards in the first place. And if it is, such users are the ones that are least likely to need assistance choosing a new community.

Now although I see your point, I'd like to interject for a moment since Free Software is about ensuring that the source code is open to all, shared for all users, with access to source code never restricted. 

It is not and never about imposing any anticommercial or antiprofit conditions: the GPL license even allows that if only paying customers are licensed to use the program, only they can request the source code without restriction, though if they have users of their own those people will also have the right to request source code from them.

**Comment by MightyFo0l** [comment]
I think those 'high quality posters' you're talking about will have or had a 4chan pass for years. At least I do. It might be handy posting a poll somewhere (maybe /g/?) to see if users feel really affected by this. You can't simply say users who use captcha v1 = high quality posters

**Comment by ccd0** [comment]
@MightyFo0l

> I think those 'high quality posters' you're talking about will have or had a 4chan pass for years.

It's not that high quality posters necessarily use Recaptcha v1; it's that the people who say they will be leaving 4chan for this reason are higher quality than average simply because they know about the feature at all. And I would expect anyone using 4chan X to be higher quality than average also.

> At least I do. It might be handy posting a poll somewhere (maybe /g/?) to see if users feel really affected by this.

Many people have commented that they would leave 4chan if Recaptcha v1 was removed:

https://desuarchive.org/a/thread/162913805/#162980382
https://rbt.asia/g/thread/62145497/#62147884
https://rbt.asia/g/thread/62145497/#62148007
https://rbt.asia/g/thread/62986178/#62987733
https://rbt.asia/g/thread/62986178/#62991569
https://rbt.asia/g/thread/62986178/#63002186
https://rbt.asia/g/thread/62986178/#63018702
https://rbt.asia/g/thread/62986178/#63019806
https://rbt.asia/g/thread/62999725/#63004121
https://desuarchive.org/qa/thread/1635764/#1637738
https://desuarchive.org/qa/thread/1635764/#1648554
https://rbt.asia/g/thread/63050879/#63051044
https://rbt.asia/g/thread/63050879/#63051068
https://rbt.asia/g/thread/63050879/#63051480
https://desuarchive.org/int/thread/80948415/#80950172
https://rbt.asia/g/thread/63124949/#63124949

As I've said, I expect this to be a minority of 4chan's users, but still a large number, especially compared to the size of most altchans.

@antonizoon

> And if it is, such users are the ones that are least likely to need assistance choosing a new community.

This is a fair point.

@Fasmea

>I feel it's against the spirit of free software.

The sense in which I agree with this is that we want to make sure the software is serving the users' instead of our own interests. Software that doesn't serve the users can still be free software, but it will be forked and changed so it does serve the users.

In this spirit, if I decide to go ahead with this, it should be minimally annoying and maximally informative. 4chan X is probably going to show some sort of message anyway explaining to users that Recaptcha v1 is dead; otherwise I would get a slew of needless bug reports asking why it is broken. It's not very annoying to add a short bit to the message saying something to the affect of "a list of boards welcoming affected users is [here]."

A list on a separate page like that (that is, not within 4chan X) would also be easier for people to link on 4chan for the benefit of people not using 4chan X. And it might be useful to compile such a list even if I decide not to link it from 4chan X. Organizing the list by the boards on 4chan each site would be most relevant to would still be a good idea.

We could even avoid making our own list and simply link to existing directories like http://www.allchans.org/. But I've always found those a bit lacking. We should probably link to such places anyway if we do make a captcha refugee focused list.

Another thing that could be improved along the lines of being informative rather than prescriptive is dropping any of my personal bias and having the list open to any sites, rather than just "deserving" sites or "preferably not 8ch-style board hosting services or sites with a reputation for raids on 4chan".

**Comment by ccd0** [comment]
We've got a wiki here, so we might as well get started:
https://github.com/ccd0/4chan-x/wiki/4chan-Alternatives
Anyone is free to add their site.

@mewch and @samachannel: Please add your site to the wiki page if you still want to participate.

Whether I end up linking this from 4chan X will depend on how useful the list ends up being and user reactions to the idea.

**Comment by majestrate** [comment]
I have in the past explored rebasing my ui for [nntpchan](https://github.com/majestrate/nntpchan) (decentralized imageboard of doom ™ powered by usenet) atop 4chan-x but that was a while ago and idk how much the code changes. If 4chan splits again perphaps consider a "decentralized" solution?

In all honesty I don't think that very many will migrate off 4chan because of lack of alternative captcha.

**Comment by antonizoon** [comment]
As Riot/Matrix (slack and irc replacement) and Pawoo/Mastodon (microblogging) show, federation is the killer feature that allows a viable alternative to centralized social networks to be constructed, uniting these little communities into big competitors. If there must be a successor to 4chan in the future, it will probably be using a federated protocol like NNTPChan.

@majestrate At the Bibliotheca Anonoma (we run Desuarchive and Rbt.asia and wiki.bibanon.org), we are considering a possible NNTPChan setup that could . We should probably aim to add NNTP support to multiple popular imageboard scripts such as tinyboard and infinity so that these diaspora of little imageboards can become one great alternative. Contact us on irc.rizon.net #bibanon if you're interested in this effort.

There may still be clustering (like gmail with email), and the nodes to federate with will probably have to group up to follow the laws of the nation the server is hosted in. No pizza bullshit for example, which killed the last attempt outside of tor... But hey, 4chan follows the laws of the State of California anyway.

**Comment by ctrlcctrlv** [comment]
I disagree with your prediction. Some small percentage of paranoid users might leave, but they've been leaving for years; if this is the straw that breaks the camel's back, it really makes me wonder why they ignored all the previous straws.

The majority of today's 4chan users don't care if Google tracks them; March 31 2018 will be a non-event.

**Comment by ctrlcctrlv** [comment]
@antonizoon Tinyboard and Infinity are both unmaintained abandoned branches. Please don't waste time porting NNTPchan network functionality to them. Vichan is still being maintained (by me and @czaks), that's the fork to target. Tinyboard users have an upgrade path to Vichan; Infinity users are using deprecated software and should either switch to the N. T. Technology sponsored OpenIB fork or switch to a more sane software like LynxChan.

**Comment by antonizoon** [comment]
@ctrlcctrlv I see, the imageboard engine ecosystem has progressed tremendously from when I last looked at it. Any other imageboard/textboard systems of note? I found Monaba to be awesome.

**Comment by ccd0** [comment]
Going to try reviving this idea for what it's worth now that they've killed the noscript captcha. I expect this won't be the last time captcha gets worse either.

**Comment by ctrlcctrlv** [comment]
I obviously recommend that 8kun not be considered for recommendation.
